dashbuilder-shared/dashbuilder-dataset-shared/src/main/java/org/dashbuilder/dataset/engine/function/MaxFunction.java

```diff
@@ -38,7 +38,10 @@ public Object aggregate(List values) {
             return null;
         }
 
-        List sorted = (List) values.stream().sorted().collect(Collectors.toList());
+        List sorted = (List) values.stream()
+                .filter(i -> i != null)
+                .sorted().collect(Collectors.toList());
```

Comment:
Is it really necessary to sort the entire list (O(n*log(n)) complexity just to get the max value?
Can't for example [max](http://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#max-java.util.Comparator-) to get the same result in linear time? The same comment applies to MinFunction.